### PR TITLE
fix: [CDS-46986]: integrating version api in ami artifact

### PR DESCRIPTION
**Pull request by satyam-harness**
: version api integrated for ami artifact

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx [hunk 1]
```diff
@@ -91,6 +95,56 @@ function FormComponent({
...
+        formik.setFieldValue('versionRegex', formik.values?.version)
+      } else {
+        formik.setFieldValue('versionRegex', '')
+      }
+    }
+  }, [formik.values?.version])
+
+  const getVersions = (): SelectOption[] => {
+    if (isVersionLoading) {
+      return [{ label: 'Loading Versions...', value: 'Loading Versions...' }]
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx:143]: `Loading Versions...` can be moved to strings

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AmazonMachineImageSource/AmazonMachineImageSource.tsx [hunk 2]
```diff
@@ -57,12 +73,89 @@ const Content = (props: JenkinsRenderContent): React.ReactElement => {
...
+        formik.setFieldValue('versionRegex', formik.values?.version)
+      } else {
+        formik.setFieldValue('versionRegex', '')
+      }
+    }
+  }, [formik.values?.version])
+
+  const getVersions = (): SelectOption[] => {
+    if (isVersionLoading) {
+      return [{ label: 'Loading Versions...', value: 'Loading Versions...' }]
```
Comment (PratyushGar) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AmazonMachineImageSource/AmazonMachineImageSource.tsx:154]: Same here

**Review by PratyushGar — APPROVED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx [hunk 3]
```diff
@@ -91,6 +95,56 @@ function FormComponent({
...
+        'content-type': 'application/json'
+      }
+    },
+    queryParams: {
+      accountIdentifier: accountId,
+      projectIdentifier,
+      orgIdentifier,
+      region: get(formik, 'values.spec.region'),
+      connectorRef: connectorRefValue || '',
+      versionRegex: '*'
```
Comment (Sagar-Rajput7) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx:119]: Is this versionRegex value hardcoding expected here?

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AmazonMachineImageSource/AmazonMachineImageSource.tsx [hunk 4]
```diff
@@ -57,12 +73,89 @@ const Content = (props: JenkinsRenderContent): React.ReactElement => {
...
+        'content-type': 'application/json'
+      }
+    },
+    queryParams: {
+      accountIdentifier: accountId,
+      projectIdentifier,
+      orgIdentifier,
+      region: getFinalQueryParamValue(regionValue),
+      connectorRef: getFinalQueryParamValue(connectorRefValue),
+      versionRegex: '*',
```
Comment (Sagar-Rajput7) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AmazonMachineImageSource/AmazonMachineImageSource.tsx:116]: same here

**Review by Sagar-Rajput7 — APPROVED** [review]

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx (hunk 3 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactRepository/ArtifactLastSteps/AmazonMachineImage/AmazonMachineImage.tsx:119]: this is intentional as BE api wanted this

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AmazonMachineImageSource/AmazonMachineImageSource.tsx (hunk 4 above)
Comment (satyam-harness) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AmazonMachineImageSource/AmazonMachineImageSource.tsx:116]: this is intentional as BE api wanted this

**Pull request merged**